src/torchmetrics/functional/audio/nisqa.py

```diff
@@ -0,0 +1,371 @@
+        >>> import torch
+        >>> from torchmetrics.functional.audio.nisqa import non_intrusive_speech_quality_assessment
+        >>> _ = torch.manual_seed(42)
+        >>> preds = torch.randn(16000)
+        >>> non_intrusive_speech_quality_assessment(preds, 16000)
+        tensor([1.0433, 1.9545, 2.6087, 1.3460, 1.7117])
+
+    References:
+        - [1] G. Mittag and S. Möller, "Non-intrusive speech quality assessment for super-wideband speech communication
+          networks", in Proc. ICASSP, 2019.
+        - [2] G. Mittag, B. Naderi, A. Chehadi and S. Möller, "NISQA: A deep CNN-self-attention model for
+          multidimensional speech quality prediction with crowdsourced datasets", in Proc. INTERSPEECH, 2021.
+
+    """
+    if not _LIBROSA_AVAILABLE or not _REQUESTS_AVAILABLE:
+        raise ModuleNotFoundError(
+            "NISQA metric requires that librosa and requests are installed. Install as `pip install librosa requests`."
+        )
+    model, args = _load_nisqa_model()
+    model.eval()
+    x = preds.reshape(-1, preds.shape[-1])
+    x = _get_librosa_melspec(x.cpu().numpy(), fs, args)
+    x, n_wins = _segment_specs(torch.from_numpy(x), args)
+    with torch.no_grad():
+        x = model(x, n_wins.expand(x.shape[0]))
+    # ["mos_pred", "noi_pred", "dis_pred", "col_pred", "loud_pred"]
+    # the dimensions are always listed in the papers as MOS, noisiness, coloration, discontinuity and loudness
+    # but based on original code the actual model output order is MOS, noisiness, discontinuity, coloration, loudness
+    return x.reshape(preds.shape[:-1] + (5,))
+
+
+@lru_cache
+def _load_nisqa_model() -> Tuple[nn.Module, Dict[str, Any]]:
+    """Load NISQA model and its parameters.
+
+    Returns:
+        Tuple ``(model,args)`` where ``model`` is the NISQA model and ``args`` is a dictionary with all its parameters
+
+    """
+    model_path = os.path.expanduser(os.path.join(NISQA_DIR, "nisqa.tar"))
+    if not os.path.exists(model_path):
+        _download_weights()
+    checkpoint = torch.load(model_path, map_location="cpu", weights_only=True)
+    args = checkpoint["args"]
+    model = _NISQADIM(args)
+    model.load_state_dict(checkpoint["model_state_dict"], strict=True)
+    return model, args
+
+
+def _download_weights() -> None:
+    """Download NISQA model weights."""
+    url = "https://github.com/gabrielmittag/NISQA/raw/refs/heads/master/weights/nisqa.tar"
+    nisqa_dir = os.path.expanduser(NISQA_DIR)
+    os.makedirs(nisqa_dir, exist_ok=True)
+    saveto = os.path.join(nisqa_dir, "nisqa.tar")
+    if os.path.exists(saveto):
+        return
+    rank_zero_info(f"downloading {url} to {saveto}")
+    myfile = requests.get(url)
+    with open(saveto, "wb") as f:
+        f.write(myfile.content)
+
+
+class _NISQADIM(nn.Module):
+    # main NISQA model definition
+    # ported from https://github.com/gabrielmittag/NISQA
+    # Copyright (c) 2021 Gabriel Mittag, Quality and Usability Lab
+    # MIT License
```

Comment:
@Borda we had copyright issue before, anything we need to do here?

Reply:
The model weights are also under CC BY-NC-SA license

Reply:
that is a problem :(

Reply:
we will need to get the authors to allow distribute it with apache licence

Reply:
Original code is not a pip installable package so can't really import as an optional dependency.

DNSMOS is similar: MIT license code and CC BY license weights. How was this managed? Or was this overlooked?